crates/oxc_linter/src/rules/jest/prefer_comparison_matcher.rs

```diff
@@ -0,0 +1,476 @@
+    /// expect(x < 7).not.toEqual(true);
+    /// expect(x <= y).toStrictEqual(true);
+    /// ```
+    ///
+    /// ```js /// // valid
+    /// expect(x).toBeGreaterThan(5);
+    /// expect(x).not.toBeLessThanOrEqual(7);
+    /// expect(x).toBeLessThanOrEqual(y);
+    /// // special case - see below
+    /// expect(x < 'Carl').toBe(true);
+    /// ```
+    ///
+    PreferComparisonMatcher,
+    style,
+);
+
+impl Rule for PreferComparisonMatcher {
+    fn run_once(&self, ctx: &LintContext) {
+        for possible_jest_node in &collect_possible_jest_call_node(ctx) {
+            Self::run(possible_jest_node, ctx);
+        }
+    }
+}
+
+impl PreferComparisonMatcher {
+    fn run<'a>(possible_jest_node: &PossibleJestNode<'a, '_>, ctx: &LintContext<'a>) {
+        let node = possible_jest_node.node;
+        let AstKind::CallExpression(call_expr) = node.kind() else {
+            return;
+        };
+        let Some(parse_expect_jest_fn) =
+            parse_expect_jest_fn_call(call_expr, possible_jest_node, ctx)
+        else {
+            return;
+        };
+        let Some(matcher) = parse_expect_jest_fn.matcher() else {
+            return;
+        };
+        let Some(parent_node) = parse_expect_jest_fn.head.parent else {
+            return;
+        };
+        let Expression::CallExpression(parent_call_expr) = parent_node else {
+            return;
+        };
+        let Some(Argument::Expression(Expression::BinaryExpression(binary_expr))) =
+            parent_call_expr.arguments.first()
+        else {
+            return;
+        };
+        let Some(Argument::Expression(first_matcher_arg)) = parse_expect_jest_fn.args.first()
+        else {
+            return;
+        };
+
+        if Self::is_comparing_to_string(binary_expr)
+            || !is_equality_matcher(matcher)
+            || !matches!(first_matcher_arg.get_inner_expression(), Expression::BooleanLiteral(_))
+        {
+            return;
+        }
+
+        let has_not_modifier = parse_expect_jest_fn
+            .modifiers()
+            .iter()
+            .filter(|modifier| modifier.is_name_equal("not"))
+            .count()
+            > 0;
+        let Expression::BooleanLiteral(matcher_arg_value) =
+            first_matcher_arg.get_inner_expression()
+        else {
+            return;
+        };
+        let negated = matcher_arg_value.value == has_not_modifier;
+        let preferred_matcher = Self::determine_matcher(binary_expr.operator, negated);
+        let Some(prefer_matcher_name) = preferred_matcher else {
+            return;
+        };
+
+        ctx.diagnostic_with_fix(UseToBeComparison(prefer_matcher_name, matcher.span), || {
+            let call_span_end = Span::new(binary_expr.span.end, parent_call_expr.span.end)
+                .source_text(ctx.source_text());
+            let arg_span_end = Span::new(matcher_arg_value.span.end, call_expr.span.end)
+                .source_text(ctx.source_text());
```

Comment:
Nit: can we add some comments to describe where `call_span_end` and `arg_span_end` point to? e.g.

```js
expect(1 > 2).toBe(false)
            ^
```